### DISCUSSION TITLE
Controling mass deployment with Intune

### DISCUSSION THREAD

**Discussion by rrice2004** [uBlockOrigin/uBOL-home#232] [Q&A] [answered]
We are making the switch from uBO to uBOL. We're an Intune shop and I've already made the adjustments for testing. With Intune, all i need to do is add the extension ID to the deployment and it will push it. What I've noticed is it's deploying fine but has the pop up start window and defaults to basic.

There are no options within the extension deployment options for custom options like no pop up start window and setting the blocking to Optimal. 
Is there a guide that I haven't come across how to deploy this with custom options? Currently the guides I've found are all the same as the process we are using.

**Comment by MasterKia** [comment]
> no pop up start

https://github.com/uBlockOrigin/uBOL-home/wiki/Managed-settings#disablefirstrunpage

> setting the blocking to Optimal

Please comment here:
https://github.com/uBlockOrigin/uBOL-home/discussions/35#discussioncomment-10958972

**Reply by rrice2004** [reply]
There's no option for that in the Intune deployment, that's why i started this particular thread. The only options are to insert the extension ID.

**Comment by gorhill** [comment, marked as answer]
See <https://github.com/uBlockOrigin/uBOL-home/wiki/Managed-settings>.

Regarding _Optimal_ or _Complete_ mode as the default one, it's not possible because these modes require a user gesture so that they can see and agree with the browser warning related to granting "read and modify data on all websites".

People responsible of deployment should talk to Chrome development team to get a way to grant such optional permissions as a deployment option. If so, I can then verify that uBOL installs and take into account that it is given such permission to automatically set it to _Optimal_ mode by default.

**Reply by rrice2004** [reply]
I appreciate the information. I'll have to create documentation then to send out before deployment.

**Reply by gorhill** [reply]
I just ran an experiment by setting `runtime_allowed_hosts` to `*://*` ([ref](https://support.google.com/chrome/a/answer/9867568?hl=en)) and though I am not entirely confident about the result (since I had the extension installed locally before removing it), it does seem that this setting might be able to grant heightened permission for all sites.

I modified the first-install code in uBOL and as a result it sets itself to _Optimal_ mode by default. But I feel this needs more testing to really confirm that `runtime_allowed_hosts` can dictate whether `browser.permissions.contains({ origins: [ '<all_urls>' ] })` succeed.

---

Never mind, doesn't work, I made a mistake when testing. Chrome will have to come up with a way for optional permissions to be automatically granted for deployment purpose.

**Comment by gorhill** [comment]
Related: https://github.com/uBlockOrigin/uBOL-home/issues/113
